tools/pymanifold.cpp

```diff
@@ -0,0 +1,169 @@
+      .def(py::init([](std::vector<Manifold> &manifolds) {
+        Manifold result;
+        for (Manifold &manifold : manifolds)
+          result += manifold;
+        return result;
+      }), "Construct manifold as the union of a set of manifolds.")
+      .def(py::self + py::self)
+      .def(py::self - py::self)
+      .def(py::self ^ py::self)
+      .def("transform",
+           [](Manifold &self, py::array_t<float> &mat) {
+             auto mat_view = mat.unchecked<2>();
+             if (mat_view.shape(0) != 3 || mat_view.shape(1) != 4)
+               throw std::runtime_error("Invalid matrix shape");
+             glm::mat4x3 mat_glm;
+             for (int i = 0; i < 3; i++) {
+               for (int j = 0; j < 4; j++) {
+                 mat_glm[j][i] = mat_view(i, j);
+               }
+             }
+             return self.Transform(mat_glm);
+           })
+      .def(
+          "translate",
+          [](Manifold self, float x = 0.0f, float y = 0.0f, float z = 0.0f) {
+            return self.Translate(glm::vec3(x, y, z));
+          },
+          py::arg("x") = 0.0f, py::arg("y") = 0.0f, py::arg("z") = 0.0f)
+      .def("scale", static_cast<Manifold (*)(Manifold, float)>(
+                        [](Manifold self, float scale) {
+                          return self.Scale(glm::vec3(scale));
+                        }), py::arg("scale"))
+      .def("scale", static_cast<Manifold (*)(Manifold, py::array_t<float> &)>(
+                        [](Manifold self, py::array_t<float> &scale) {
+                          auto scale_view = scale.unchecked<1>();
+                          if (scale_view.shape(0) != 3)
+                            throw std::runtime_error("Invalid vector shape");
+                          glm::vec3 v(scale_view(0), scale_view(1),
+                                      scale_view(2));
+                          return self.Scale(v);
+                        }), py::arg("v"))
+      .def("rotate",
+           [](Manifold self, float xDegrees = 0.0f, float yDegrees = 0.0f, float zDegrees = 0.0f) {
+             return self.Rotate(xDegrees, yDegrees, zDegrees);
+           }, py::arg("xDegrees") = 0.0f, py::arg("yDegrees") = 0.0f, py::arg("zDegrees") = 0.0f)
+      .def("export",
+           [](Manifold &self, std::string name) { exportManifold(self, name); },
+        py::arg("filename"),
+           "Export the manifold object to file, where the file type is determined from file extension.")
+      .def("warp",
+           [](Manifold self, const std::function<Float3(Float3)> &f) {
+             return self.Warp([&f](glm::vec3 &v) {
+               Float3 fv = f(std::make_tuple(v.x, v.y, v.z));
+               v.x = std::get<0>(fv);
+               v.y = std::get<1>(fv);
+               v.z = std::get<2>(fv);
+             });
+           }, py::arg("f"))
+      .def("refine", [](Manifold self, int n) { return self.Refine(n); }, py::arg("n"))
+      .def_static(
+          "FromMesh",
```

Comment:
Seems like we should add a "ToMesh" method for symmetry. This will allow the bindings to work with the likes of Blender. Still, they'll also need the `MeshRelation` and the rest of the API in order to manage their texture coordinates and material properties.

Reply:
and the "Smooth" method, which I consider as one of the main selling points of this lib !

Reply:
With this merged, it should be fairly easy to expand the Python API; would you like to try adding `Smooth`?

Reply:
`Smooth` is already added in line 94.

Reply:
Thanks! Totally missed that.